### PR TITLE
[Rllib] Fix test restore checkpoint with nested obs space for RL Modules

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

With RL Modules, we are unable to instantiate Policies without the MARL Module and policy id logic that resides in rollout worker. This PR makes it so that we get around constructing our own policy in `test_restore_checkpoint_with_nested_obs_space` until this is fixed.

### CODE REVIEW

File: rllib/policy/tests/test_policy_checkpoint_restore.py [hunk 1]
```diff
@@ -103,7 +101,22 @@ def test_restore_checkpoint_with_nested_obs_space(self):
             space.original_space = gym.spaces.Discrete(2)
             space = space.original_space
 
-        policy = PPOTorchPolicy(obs_space, gym.spaces.Discrete(2), MODEL_DEFAULTS)
+        class NestedObsSpaceDummyEnv(gym.Env):
```
Comment (kouroshHakha) [inline rllib/policy/tests/test_policy_checkpoint_restore.py:104]: This is not needed, right? You can directly pass obs_space, and action_space to config without an env?
```
policy = PPOConfig().environment(observation_space=obs_space, action_space=...).build().get_policy()
```

If env started complaining, turn the env checker off.

**Comment by ArturNiederfahrenhorst** [comment]
Screenshot 2023-03-30 at 14 06 49

Works

**Comment by ArturNiederfahrenhorst** [comment]
Gonna remove the rl modules flags

**Review by kouroshHakha — APPROVED** [review]

**Comment by kouroshHakha** [comment]
Good to merge. but not 2.4 critical.

**Pull request merged**